src/main/java/org/testng/internal/Invoker.java

```diff
@@ -1281,8 +1281,13 @@ private void handleInvocationResults(
     int status =
         computeTestStatusComparingTestResultAndStatusHolder(testResult, holder, wasResultUnaltered);
     boolean handled = holder.handled;
+    IRetryAnalyzer retryAnalyzer = testMethod.getRetryAnalyzer(testResult);
+
+    //For backward compatibility.
```

Comment:
What I said previously: if the default method is well implemented, you don't have to do that ;)

Reply:
True that. Fixed.